### PR TITLE
Add org.kuchelmeister.toolbxtuner

### PR DESCRIPTION
**Pull request by 13hannes11**
### Please confirm your submission meets all the criteria

- [ ] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/13hannes11/toolbx-tuner
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

**Permission requirements:**

* The application cannot really be sandboxed as it needs access to the hosts commandline to manipulate `podman` and containertoolbxes. 
* Access to the users home folder is required to allow modifications in `.bashrc.d`. This allows a change of home directory for toolboxes.

### CODE REVIEW

**Comment by 13hannes11** [comment]
bot, build org.kuchelmeister.toolbxtuner

File: org.kuchelmeister.toolbxtuner.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--talk-name=org.freedesktop.Flatpak"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
+        ],
```
Comment (hfiguiere) [inline org.kuchelmeister.toolbxtuner.json:22]: this is not allowed. You need to use flatpak-builder-tools to generate a manifest for the cargo dependencies if they are not vendored.

File: org.kuchelmeister.toolbxtuner.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "toolbx-tuner",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/13hannes11/toolbx-tuner.git",
```
Comment (hfiguiere) [inline org.kuchelmeister.toolbxtuner.json:47]: you need a commit or tag.

Comment (hfiguiere) [reply org.kuchelmeister.toolbxtuner.json:47]:
```suggestion
                    "url" : "https://github.com/13hannes11/toolbx-tuner.git"
```

also JSON syntax error.

File: org.kuchelmeister.toolbxtuner.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        "--talk-name=org.freedesktop.Flatpak"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
+        ],
+        "env" : {
+            "RUST_BACKTRACE" : "1",
+            "RUST_LOG" : "toolbx-tuner=debug"
```
Comment (hfiguiere) [inline org.kuchelmeister.toolbxtuner.json:25]: debug log in a release?

File: org.kuchelmeister.toolbxtuner.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    "modules" : [
+        {
+            "name" : "toolbx-tuner",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/13hannes11/toolbx-tuner.git",
+                    "branch" : "main"
```
Comment (hfiguiere) [inline org.kuchelmeister.toolbxtuner.json:48]: not a branch. a commit or a tag.

**Comment by JakobDev** [comment]
> Access to the users home folder is required to allow modifications in .bashrc.d. This allows a change of home directory for toolboxes.

Then why did you give permission for home and not only for .bashrc.d?

From your [GitHub Repo](https://github.com/13hannes11/toolbx-tuner):

> The project is currently only a user-interface concept. The program is not actually usable.

Please keep in mind that Flathub is only for full working Applications. Is your App is not finished yet, you should not put on Flathub until it is ready.

**Comment by 13hannes11** [comment]
> Please keep in mind that Flathub is only for full working Applications. Is your App is not finished yet, you should not put on Flathub until it is ready.

I was not planning to merge the pull request yet just wanted to clear hurdles in advance.

File: org.kuchelmeister.toolbxtuner.json (hunk 1 above)
Comment (13hannes11) [reply org.kuchelmeister.toolbxtuner.json:22]: Is there any recommendations for an ideal way to do that? Using flatpak-builder-tools results in a rather large manifest.

Comment (JakobDev) [reply org.kuchelmeister.toolbxtuner.json:22]: This is the recommend way. You can include other files (like the one generated from flatpak-builder-tools) in your main file.

Comment (hfiguiere) [reply org.kuchelmeister.toolbxtuner.json:22]: 1. said manifest being autogenerated that's not a problem (it's meant to just be included, and only updated with the same tools)
2. you can reduce your dependencies
3. you can use a tarball with vendored crates, which I think the default meson templates do.

**Pull request closed without merging**